api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2649,47 +2731,51 @@ public void composeObject(
 
     if (partsCount == 1) {
       ComposeSource src = sources.get(0);
-      if (headerMap == null) {
-        headerMap = new HashMap<>();
-      }
+      Multimap<String, String> headers = HashMultimap.create();
+      headers.putAll(args.extraHeaders());
+      headers.putAll(args.headers);
       if ((src.offset() != null) && (src.length() == null)) {
-        headerMap.put("x-amz-copy-source-range", "bytes=" + src.offset() + "-");
+        headers.put("x-amz-copy-source-range", "bytes=" + src.offset() + "-");
       }
 
       if ((src.offset() != null) && (src.length() != null)) {
-        headerMap.put(
+        headers.put(
             "x-amz-copy-source-range",
             "bytes=" + src.offset() + "-" + (src.offset() + src.length() - 1));
       }
-      copyObject(
-          bucketName,
-          objectName,
-          headerMap,
-          sse,
-          src.bucketName(),
-          src.objectName(),
-          src.ssec(),
-          src.copyConditions());
-      return;
-    }
-
-    Map<String, String> sseHeaders = null;
-    if (sse != null) {
-      sseHeaders = sse.headers();
-      if (headerMap == null) {
-        headerMap = new HashMap<>();
-      }
+      return copyObject(
```

Comment:
`copyObject()` only works with one source and without range. You would need to fix it accordingly.